### PR TITLE
javadocs and moved RecipeFilter and added support for it in SoapUI executions

### PR DESCRIPTION
**Pull request by olensmar**
i'll continue pushing javadoc updates to this...

### CODE REVIEW

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/TestRecipeBuilder.java [hunk 1]
```diff
@@ -80,18 +99,26 @@ public TestRecipeBuilder addStep(TestStepBuilder testStepBuilder) {
...
-     * @param filePath Certificate file path
+     * @param filePath Certificate file path on server
      */
     public TestRecipeBuilder withClientCertificate(String filePath) {
         testCase.setClientCertFileName(filePath);
         return this;
     }
 
+    /**
+     * @param password password for certificate configured on server
```
Comment (spindelmanne) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/TestRecipeBuilder.java]: The password isn't configured on the server if you're attaching it, is it?

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/AssertionNames.java [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-    static final String JSON_PATH_COUNT = "JsonPath Count";
-    static final String GROOVY_SCRIPT = "Script Assertion";
-    static final String RESPONSE_SLA = "Response SLA";
-    static final String JDBC_STATUS = "JDBC Status";
-    static final String JDBC_TIME_OUT = "JDBC Timeout";
-    static final String SCHEMA_COMPLIANCE = "Schema Compliance";
-    static final String SOAP_FAULT = "SOAP Fault";
-    static final String NOT_SOAP_FAULT = "Not SOAP Fault";
+/**
+ * Constants for all types of assertions
```
Comment (spindelmanne) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/AssertionNames.java]: I suggest:

"Constants containing the names of all assertions."

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/attachments/Attachments.java [hunk 3]
```diff
@@ -12,6 +10,10 @@
 import java.net.URLConnection;
 import java.nio.file.Files;
 
+/**
+ * Utility class for creating a RequestAttachmentBuilder for various types of attachments
```
Comment (spindelmanne) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/attachments/Attachments.java]: A bit too technical perhaps. I suggest:

"Utility class for adding attachments to a request"

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/auth/Authentications.java [hunk 4]
```diff
@@ -1,5 +1,9 @@
 package com.smartbear.readyapi4j.auth;
 
+/**
+ * Utility class for creating AuthenticationBuilders
```
Comment (spindelmanne) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/auth/Authentications.java]: Maybe "Utility class for adding authentication to a request"?

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/execution/Execution.java [hunk 5]
```diff
@@ -19,7 +19,15 @@
...
     RecipeExecutionResult getExecutionResult();
 
+    /**
+     * Gets a list of error messages collected during this execution
+     */
+
     List<String> getErrorMessages();
 
+    /**
+     * Cancels this execution if it is still ongoing
```
Comment (spindelmanne) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/execution/Execution.java]: ongoing -> running

File: readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java [hunk 6]
```diff
@@ -17,6 +17,16 @@
 
 import static com.smartbear.readyapi4j.support.AssertionUtils.assertExecution;
 
+/**
+ * Utility for running recipes either locally or on a remote TestServer based on system/env property values<br/>
+ * <p>
+ * If a <code>testserver.endpoint</code> system or environment property is available and contains a valid URL
+ * remote execution will be used by the executeRecipe method. Corresponding
```
Comment (spindelmanne) [inline readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java]: "Corresponding ..." -> "The values of the ..."

File: readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java [hunk 7]
```diff
@@ -17,6 +17,16 @@
...
+ * <code>testserver.user</code> and <code>testserver.password</code> properties will be used to
```
Comment (spindelmanne) [inline readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java]: properties -> properties/environment variables

File: readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java [hunk 8]
```diff
@@ -27,6 +37,14 @@
 
     private static RecipeExecutor executor;
 
+    /**
+     * Builds and executes a TestRecipe from the specified TestStepBuilders using either a local or remote
+     * executor as configured. Execution is always done synchronously.
```
Comment (spindelmanne) [inline readyapi4j-facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java]: Replace the last sentence with: "The recipe is always executed synchronously"

File: readyapi4j-local/src/main/java/com/smartbear/readyapi4j/local/execution/SoapUIRecipeExecutor.java [hunk 9]
```diff
@@ -34,15 +35,22 @@
...
+    private final List<RecipeFilter> recipeFilters = new CopyOnWriteArrayList<>();
     private ObjectMapper objectMapper;
 
     @Override
     public Execution submitRecipe(TestRecipe recipe) {
+        notifyRecipeFilters( recipe );
         return postTestCase(recipe.getTestCase(), true);
     }
 
+    private void notifyRecipeFilters(TestRecipe recipe) {
```
Comment (spindelmanne) [inline readyapi4j-local/src/main/java/com/smartbear/readyapi4j/local/execution/SoapUIRecipeExecutor.java]: applyRecipeFilters would be a better name

**Comment by spindelmanne** [comment]
Looks good now! Merging

**Comment by olensmar** [comment]
thanks @spindelmanne and @prakashmrec for moving this forward

**Pull request merged**